ProjectReference/@ExcludeFromNuPkg=true with EmbedSourceFiles=true breaks build

After setting a project reference's ExcludeFromNuPkg metadata to true, on a project that embeds source files, the build breaks with this error:

> NuProj.0.10.10-beta-g13363167e1\tools\NuProj.targets(962,5): error MSB4044: The "ReadPdbSourceFiles" task was not given a value for the required parameter "PdbPath". 

I also notice that ExcludeFromNuPkg isn't exposed via the XAML files.

--- a/src/NuProj.Targets/NuProj.targets
+++ b/src/NuProj.Targets/NuProj.targets
@@ -959,7 +959,8 @@
       <_PdbFiles Include="@(PackageFile)"
                  Condition="'%(Extension)' == '.pdb'" />
     </ItemGroup>
-    <ReadPdbSourceFiles PdbPath="%(_PdbFiles.Identity)">
+    <ReadPdbSourceFiles PdbPath="%(_PdbFiles.Identity)"
+                        Condition=" '@(_PdbFiles)' != '' ">
       <Output TaskParameter="SourcePaths"
               ItemName="_SourceFile" />
     </ReadPdbSourceFiles>

--- a/src/NuProj.Tests/ExtensibilityTests.cs
+++ b/src/NuProj.Tests/ExtensibilityTests.cs
@@ -4,9 +4,17 @@
     using System.Threading.Tasks;
     using NuProj.Tests.Infrastructure;
     using Xunit;
+    using Xunit.Abstractions;
 
     public class ExtensibilityTests
     {
+        private ITestOutputHelper logger;
+
+        public ExtensibilityTests(ITestOutputHelper logger)
+        {
+            this.logger = logger;
+        }
+
         [Fact]
         public async Task Extensibility_LoadsDotUserFile()
         {
@@ -14,7 +22,7 @@
                 "Extensibility_LoadsDotUserFile",
                 @"NuGetPackage\NuGetPackage.nuproj");
             var properties = MSBuild.Properties.Default;
-            var result = await MSBuild.ExecuteAsync(projectPath, "Rebuild", properties);
+            var result = await MSBuild.ExecuteAsync(projectPath, "Rebuild", properties, this.logger);
             result.AssertSuccessfulBuild();
 
             var warnings = result.WarningEvents.First();

--- a/src/NuProj.Tests/Infrastructure/MSBuild.cs
+++ b/src/NuProj.Tests/Infrastructure/MSBuild.cs
@@ -10,20 +10,21 @@ using Microsoft.Build.Framework;
 using Microsoft.Build.Logging;
 
 using Xunit;
+using Xunit.Abstractions;
 
 namespace NuProj.Tests.Infrastructure
 {
     public static class MSBuild
     {
-        public static Task<BuildResultAndLogs> RebuildAsync(string projectPath, string projectName = null, IDictionary<string, string> properties = null)
+        public static Task<BuildResultAndLogs> RebuildAsync(string projectPath, string projectName = null, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
             var target = string.IsNullOrEmpty(projectName) ? "Rebuild" : projectName.Replace('.', '_') + ":Rebuild";
-            return MSBuild.ExecuteAsync(projectPath, new[] { target }, properties);
+            return MSBuild.ExecuteAsync(projectPath, new[] { target }, properties, testLogger);
         }
 
-        public static Task<BuildResultAndLogs> ExecuteAsync(string projectPath, string targetToBuild, IDictionary<string, string> properties = null)
+        public static Task<BuildResultAndLogs> ExecuteAsync(string projectPath, string targetToBuild, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
-            return MSBuild.ExecuteAsync(projectPath, new[] { targetToBuild }, properties);
+            return MSBuild.ExecuteAsync(projectPath, new[] { targetToBuild }, properties, testLogger);
         }
 
         /// <summary>
@@ -33,7 +34,7 @@ namespace NuProj.Tests.Infrastructure
         /// <param name="targetsToBuild">The targets to build. If not specified, the project's default target will be invoked.</param>
         /// <param name="properties">The optional global properties to pass to the project. May come from the <see cref="MSBuild.Properties"/> static class.</param>
         /// <returns>A task whose result is the result of the build.</returns>
-        public static async Task<BuildResultAndLogs> ExecuteAsync(string projectPath, string[] targetsToBuild = null, IDictionary<string, string> properties = null)
+        public static async Task<BuildResultAndLogs> ExecuteAsync(string projectPath, string[] targetsToBuild = null, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
             targetsToBuild = targetsToBuild ?? new string[0];
 
@@ -44,6 +45,7 @@ namespace NuProj.Tests.Infrastructure
                 Loggers = new List<ILogger>
                 {
                     new ConsoleLogger(LoggerVerbosity.Detailed, logLines.Add, null, null),
+                    new ConsoleLogger(LoggerVerbosity.Minimal, v => testLogger?.WriteLine(v.TrimEnd()), null, null),
                     logger,
                 },
             };

--- a/src/NuProj.Tests/Infrastructure/Scenario.cs
+++ b/src/NuProj.Tests/Infrastructure/Scenario.cs
@@ -6,6 +6,7 @@ using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 
 using NuGet;
+using Xunit.Abstractions;
 
 namespace NuProj.Tests.Infrastructure
 {
@@ -18,9 +19,9 @@ namespace NuProj.Tests.Infrastructure
         /// <param name="packageId">The leaf name of the project to be built or rebuilt, and the package ID to return after the build.</param>
         /// <param name="properties">Build properties to pass to MSBuild.</param>
         /// <returns>The single built package, or the package whose ID matches <paramref name="packageId"/>.</returns>
-        public static async Task<IPackage> RestoreAndBuildSinglePackageAsync([CallerMemberName] string scenarioName = null, string packageId = null, IDictionary<string, string> properties = null)
+        public static async Task<IPackage> RestoreAndBuildSinglePackageAsync([CallerMemberName] string scenarioName = null, string packageId = null, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
-            var packages = await RestoreAndBuildPackagesAsync(scenarioName, packageId, properties);
+            var packages = await RestoreAndBuildPackagesAsync(scenarioName, packageId, properties, testLogger);
             return packageId == null
                     ? packages.Single()
                     : packages.Single(p => string.Equals(p.Id, packageId, StringComparison.OrdinalIgnoreCase));
@@ -33,14 +34,14 @@ namespace NuProj.Tests.Infrastructure
         /// <param name="projectName">The leaf name of the project to be built or rebuilt.</param>
         /// <param name="properties">Build properties to pass to MSBuild.</param>
         /// <returns>A collection of the built packages.</returns>
-        public static async Task<IReadOnlyCollection<IPackage>> RestoreAndBuildPackagesAsync([CallerMemberName] string scenarioName = null, string projectName = null, IDictionary<string, string> properties = null)
+        public static async Task<IReadOnlyCollection<IPackage>> RestoreAndBuildPackagesAsync([CallerMemberName] string scenarioName = null, string projectName = null, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
             var projectFullPath = Assets.GetScenarioSolutionPath(scenarioName);
 
             var projectDirectory = Path.GetDirectoryName(projectFullPath);
             await NuGetHelper.RestorePackagesAsync(projectDirectory);
 
-            var result = await MSBuild.RebuildAsync(projectFullPath, projectName, properties);
+            var result = await MSBuild.RebuildAsync(projectFullPath, projectName, properties, testLogger);
             result.AssertSuccessfulBuild();
 
             return NuPkg.GetPackages(projectDirectory);
@@ -52,14 +53,14 @@ namespace NuProj.Tests.Infrastructure
         /// <param name="scenarioName">The name of the scenario to build. If omitted, it will be the name of the calling method.</param>
         /// <param name="projectName">The leaf name of the project to be built or rebuilt.</param>
         /// <param name="properties">Build properties to pass to MSBuild.</param>
-        public static async Task RestoreAndFailBuildAsync([CallerMemberName] string scenarioName = null, string projectName = null, IDictionary<string, string> properties = null)
+        public static async Task RestoreAndFailBuildAsync([CallerMemberName] string scenarioName = null, string projectName = null, IDictionary<string, string> properties = null, ITestOutputHelper testLogger = null)
         {
             var projectFullPath = Assets.GetScenarioSolutionPath(scenarioName);
 
             var projectDirectory = Path.GetDirectoryName(projectFullPath);
             await NuGetHelper.RestorePackagesAsync(projectDirectory);
 
-            var result = await MSBuild.RebuildAsync(projectFullPath, projectName, properties);
+            var result = await MSBuild.RebuildAsync(projectFullPath, projectName, properties, testLogger);
 
             result.AssertUnsuccessfulBuild();
         }

--- a/src/NuProj.Tests/ReferenceTests.cs
+++ b/src/NuProj.Tests/ReferenceTests.cs
@@ -8,11 +8,26 @@ using Microsoft.Build.Framework;
 using NuGet;
 using NuProj.Tests.Infrastructure;
 using Xunit;
+using Xunit.Abstractions;
 
 namespace NuProj.Tests
 {
     public class ReferenceTests
     {
+        private readonly ITestOutputHelper logger;
+
+        public ReferenceTests(ITestOutputHelper logger)
+        {
+            this.logger = logger;
+        }
+
+        [Fact]
+        public async Task References_ExcludeFromNuPkgAndEmbedSourceFiles()
+        {
+            var package = await Scenario.RestoreAndBuildSinglePackageAsync(testLogger: this.logger);
+            Assert.Empty(package.GetLibFiles());
+        }
+
         [Fact]
         public async Task References_AssignProjectConfiguration()
         {

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/Class1.cs
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/Class1.cs
@@ -1,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassLibrary39
+{
+    public class Class1
+    {
+    }
+}

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/ClassLibrary39.csproj
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/ClassLibrary39.csproj
@@ -1,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>5dac61c2-5fa3-49d0-b7a4-3722d0caf4ce</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>ClassLibrary39</RootNamespace>
+    <AssemblyName>ClassLibrary39</AssemblyName>
+    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="System"/>
+    
+    <Reference Include="System.Core"/>
+    <Reference Include="System.Xml.Linq"/>
+    <Reference Include="System.Data.DataSetExtensions"/>
+    
+    
+    <Reference Include="Microsoft.CSharp"/>
+    
+    <Reference Include="System.Data"/>
+    
+    <Reference Include="System.Net.Http"/>
+    
+    <Reference Include="System.Xml"/>
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="Class1.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
+  -->
+
+ </Project>

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/Properties/AssemblyInfo.cs
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/ClassLibrary39/Properties/AssemblyInfo.cs
@@ -1,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("ClassLibrary39")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("ClassLibrary39")]
+[assembly: AssemblyCopyright("Copyright ©  2015")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible 
+// to COM components.  If you need to access a type in this assembly from 
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("5dac61c2-5fa3-49d0-b7a4-3722d0caf4ce")]
+
+// Version information for an assembly consists of the following four values:
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/NuGetPackage/NuGetPackage.nuproj
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/NuGetPackage/NuGetPackage.nuproj
@@ -1,0 +1,47 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|AnyCPU">
+      <Configuration>Debug</Configuration>
+      <Platform>AnyCPU</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|AnyCPU">
+      <Configuration>Release</Configuration>
+      <Platform>AnyCPU</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>30bd5f2c-c93f-4b4e-aabe-9e42ec48ca9c</ProjectGuid>
+  </PropertyGroup>
+  <PropertyGroup>
+    <NuProjPath Condition=" '$(NuProjPath)' == '' ">$(MSBuildExtensionsPath)\NuProj\</NuProjPath>
+  </PropertyGroup>
+  <Import Project="$(NuProjPath)\NuProj.props" Condition="Exists('$(NuProjPath)\NuProj.props')" />
+  <PropertyGroup Label="Configuration">
+    <Id>NuGetPackage1</Id>
+    <Version>1.0.0</Version>
+    <Title>NuGetPackage1</Title>
+    <Authors>andarno</Authors>
+    <Owners>andarno</Owners>
+    <Summary>NuGetPackage1</Summary>
+    <Description>NuGetPackage1</Description>
+    <ReleaseNotes>
+    </ReleaseNotes>
+    <ProjectUrl>
+    </ProjectUrl>
+    <LicenseUrl>
+    </LicenseUrl>
+    <Copyright>Copyright © andarno</Copyright>
+    <Tags>NuGetPackage1</Tags>
+    <EmbedSourceFiles>true</EmbedSourceFiles>
+  </PropertyGroup>
+  <ItemGroup>
+    <Content Include="Readme.txt" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\ClassLibrary39\ClassLibrary39.csproj">
+      <ExcludeFromNuPkg>true</ExcludeFromNuPkg>
+    </ProjectReference>
+  </ItemGroup>
+  <Import Project="$(NuProjPath)\NuProj.targets" />
+</Project>

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/NuGetPackage/Readme.txt
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/NuGetPackage/Readme.txt
@@ -1,0 +1,1 @@
+﻿This is your NuGet package.

--- a/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/References_ExcludeFromNuPkgAndEmbedSourceFiles.sln
+++ b/src/NuProj.Tests/Scenarios/References_ExcludeFromNuPkgAndEmbedSourceFiles/References_ExcludeFromNuPkgAndEmbedSourceFiles.sln
@@ -1,0 +1,28 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 14
+VisualStudioVersion = 14.0.24711.0
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FF286327-C783-4F7A-AB73-9BCBAD0D4460}") = "NuGetPackage", "NuGetPackage\NuGetPackage.nuproj", "{30BD5F2C-C93F-4B4E-AABE-9E42EC48CA9C}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClassLibrary39", "ClassLibrary39\ClassLibrary39.csproj", "{5DAC61C2-5FA3-49D0-B7A4-3722D0CAF4CE}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{30BD5F2C-C93F-4B4E-AABE-9E42EC48CA9C}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{30BD5F2C-C93F-4B4E-AABE-9E42EC48CA9C}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{30BD5F2C-C93F-4B4E-AABE-9E42EC48CA9C}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{30BD5F2C-C93F-4B4E-AABE-9E42EC48CA9C}.Release|Any CPU.Build.0 = Release|Any CPU
+		{5DAC61C2-5FA3-49D0-B7A4-3722D0CAF4CE}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{5DAC61C2-5FA3-49D0-B7A4-3722D0CAF4CE}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{5DAC61C2-5FA3-49D0-B7A4-3722D0CAF4CE}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{5DAC61C2-5FA3-49D0-B7A4-3722D0CAF4CE}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+EndGlobal

